tools/analysis/diagnostic.go

```diff
@@ -28,11 +28,23 @@ type SuggestedFix = errors.SuggestedFix[ast.TextEdit]
 
 type TextEdit = ast.TextEdit
 
+type Severity uint8
+
+const (
+	SeverityUnknown Severity = iota
+	SeverityInfo
+	SeverityError
+	SeverityWarning
+)
+
 type Diagnostic struct {
 	Location         common.Location
 	Category         string
+	Severity         Severity
 	Message          string
 	SecondaryMessage string
+	Code             string // Diangostic code (e.g. "unused-variable")
+	DocURL           string // URL to documentation
```

Comment:
Maybe name this just `URL` [like in Go's `analysis.Diagnostic`](https://pkg.go.dev/golang.org/x/tools/go/analysis#Diagnostic) (which this package is heavily inspired by).